javatests/arcs/android/storage/service/StorageServiceNgImplTest.kt

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.storage.service
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.android.crdt.ParcelableCrdtType
+import arcs.android.storage.service.testing.FakeMessageCallback
+import arcs.android.storage.service.testing.FakeStorageChannelCallback
+import arcs.android.storage.toParcelToByteArray
+import arcs.core.data.CountType
+import arcs.core.storage.FixedDriverFactory
+import arcs.core.storage.StorageKey
+import arcs.core.storage.StorageKeyParser
+import arcs.core.storage.StoreOptions
+import arcs.core.storage.UntypedProxyMessage
+import arcs.core.storage.keys.RamDiskStorageKey
+import arcs.core.storage.testutil.MockDriverProvider
+import arcs.core.storage.testutil.testWriteBackProvider
+import arcs.core.type.Type
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class StorageServiceNgImplTest {
+
+  private val driverFactory = FixedDriverFactory(MockDriverProvider())
+  private val onProxyMessageCallback: suspend (StorageKey, UntypedProxyMessage) -> Unit =
+    { _: StorageKey, _: UntypedProxyMessage -> }
+  private val messageCallback = FakeMessageCallback()
+  private val channelCallback = FakeStorageChannelCallback()
+
+  private val scope = TestCoroutineScope()
+  private lateinit var storageService: StorageServiceNgImpl
+
+  @Before
+  fun setUp() {
+    StorageKeyParser.addParser(RamDiskStorageKey)
+    storageService = StorageServiceNgImpl(
+      scope,
+      driverFactory,
+      ::testWriteBackProvider,
+      null,
+      onProxyMessageCallback
+    )
+  }
+
+  @After
+  fun tearDown() {
+    scope.cleanupTestCoroutines()
+  }
+
+  @Test
+  fun openStorageChannel_invokesOnCreateCallback_andCreatesChannel() = scope.runBlockingTest {
+    val encodedStoreOptions = createEncodedStoreOptions(
+      RamDiskStorageKey("myCount"),
+      CountType(),
+      ParcelableCrdtType.Count
+    )
+
+    storageService.openStorageChannel(encodedStoreOptions, channelCallback, messageCallback)
+
+    // `waitForOnCreate` ensures the `onCreate` callback was invoked.
```

Comment:
This seems self-evident to me, just from the name of the method: `waitForOnCreate`. I'd suggest removing the comment